src/main/java/com/urbancode/ds/jenkins/plugins/serenarapublisher/UrbanDeployPublisher.java

```diff
@@ -107,6 +107,7 @@
  * com.urbancode.ds.jenkins.plugins.serenarapublisher.UrbanDeploySite}. </p>
  */
 public class UrbanDeployPublisher extends Notifier {
+    private final Pattern pattern = Pattern.compile("\\$\\{[^}]*}");

```

Comment:
Should be static
Otherwise...
* It will be compiled multiple times
* It will be persisted on the disk